From e67ba4491b64699ea12df02b03f91a444032b7d6 Mon Sep 17 00:00:00 2001
From: Josh Stone <cuviper@gmail.com>
Date: Wed, 15 Jan 2020 10:25:00 -0800
Subject: [PATCH] Indirectly update to libm 0.2

Unfortunately, `libm 0.2` removed the `F32Ext` and `F64Ext` extension
traits, which makes it harder to choose the different methods. However,
this was already dealt with in rust-num/num-traits#144 for `Float`, so
we can piggy-back on that here, no longer using `libm` directly.
---
 src/general/complex.rs | 10 +++-------
 src/general/real.rs    |  8 ++------
 2 files changed, 5 insertions(+), 13 deletions(-)

diff --git a/src/general/complex.rs b/src/general/complex.rs
index ad3ffe18175a..73abd514a803 100644
--- a/src/general/complex.rs
+++ b/src/general/complex.rs
@@ -7,11 +7,7 @@ use std::{f32, f64};
 use crate::general::{Field, JoinSemilattice, MeetSemilattice, RealField, SubsetOf, SupersetOf};
 
 #[cfg(not(feature = "std"))]
-use libm::F32Ext;
-#[cfg(not(feature = "std"))]
-use libm::F64Ext;
-#[cfg(not(feature = "std"))]
-use num;
+use num::Float;
 //#[cfg(feature = "decimal")]
 //use decimal::d128;
 
@@ -474,8 +470,8 @@ macro_rules! impl_complex(
 
 #[cfg(not(feature = "std"))]
 impl_complex!(
-    f32, f32, F32Ext;
-    f64, f64, F64Ext
+    f32, f32, Float;
+    f64, f64, Float
 );
 
 #[cfg(feature = "std")]
diff --git a/src/general/real.rs b/src/general/real.rs
index d1835779f5f6..9e29031c7a95 100644
--- a/src/general/real.rs
+++ b/src/general/real.rs
@@ -6,11 +6,7 @@ use approx::{RelativeEq, UlpsEq};
 use crate::general::{ComplexField, Lattice};
 
 #[cfg(not(feature = "std"))]
-use libm::F32Ext;
-#[cfg(not(feature = "std"))]
-use libm::F64Ext;
-#[cfg(not(feature = "std"))]
-use num;
+use num::Float;
 //#[cfg(feature = "decimal")]
 //use decimal::d128;
 
@@ -180,7 +176,7 @@ macro_rules! impl_real(
 );
 
 #[cfg(not(feature = "std"))]
-impl_real!(f32,f32,F32Ext; f64,f64,F64Ext);
+impl_real!(f32,f32,Float; f64,f64,Float);
 #[cfg(feature = "std")]
 impl_real!(f32,f32,f32; f64,f64,f64);
 //#[cfg(feature = "decimal")]
-- 
2.24.1

